src/shared/assets.ts

```diff
@@ -344,6 +344,33 @@ export class AssetGenerator {
 
         return result;
     }
+
+    public getAssetsPathAndProgram(): [string, string] {
+        let assetsPath = ``;
+        this.executableProjects.forEach((project) => {
+            if (project.isWebAssemblyProject) {
+                assetsPath += path.join(path.dirname(project.outputPath), path.sep);
+                assetsPath += ';';
+            }
+        });
+        assetsPath = assetsPath.slice(0, -1);
+        return [assetsPath, this.executableProjects[0].outputPath];
+    }
+
+    public isDotNet9OrNewer(): boolean {
+        let ret = false;
+        this.executableProjects.forEach((project) => {
+            if (project.isWebAssemblyProject) {
+                const projectFileText = fs.readFileSync(project.projectPath, 'utf8');
+                ret =
+                    projectFileText.toLowerCase().indexOf('net8.0') < 0 &&
```

Comment:
I feel like we should be using an XML parser, or at least a regex to try and find `<TargetFramework>` or `<TargetFramworks>` here rather than just searching blindly in the entire document for these strings.

Reply:
I wonder if it would be worth just asking the language service team to return this, as they already have it.

Reply:
Roslyn code for this: https://github.com/search?q=repo%3Adotnet%2Froslyn+targetsDotnetCore&type=code